### ISSUE TITLE
With the new feature 'global using' (C# 10) causes problem for the code generator

### ISSUE THREAD

**Issue by jan-johansson-mr** [dotnet/orleans#7284]
Hi,

The code generator is having problem with source code created with C# 10 (Visual Studio 2022 preview). The problem seem to be that the code generator depends on the **`using System.Threading.Tasks`** declaration in the file.

The **`System.Threading.Tasks`** namespace is implicitly imported with C# 10, so declarations with `Task` works without the explicit **`using System.Threading.Tasks`** (but now the code generator generates an error because the namespace was missing).

For example:
```
using Orleans;

namespace Test
{
    public interface IMyGrain : IGrainWithStringKey
    {
        Task<long> Add(long a, long b);
    }
}
```
Displays the error:
`Exc level 0: System.InvalidOperationException: Grain interface Test.IMyGrain has method Test.IMyGrain.Add(long, long) which returns a non-awaitable type Task<long>. All grain interface methods must return awaitable types. Did you mean to return Task<Task<long>>?`

The workaround is simply to declare the namespace explicitly:

```
using Orleans;

#pragma warning disable IDE0005
// Orleans code generator need this declaration ('System.Threading.Tasks' is included implicitly by compiler with C# 10)
using System.Threading.Tasks;

namespace Test
{
    public interface IMyGrain : IGrainWithStringKey
    {
        Task<long> Add(long a, long b);
    }
}
```
The disable warning pragma is used to get rid of intellisense suggestion about removing the `using` (the code cleanup removes the `using` statement anyway though...).

**Comment by ReubenBond** [comment]
I assume this is with Orleans 3.5.0. Support for unreleased language features wont work until vNext, where source generators are used, since otherwise that would require Orleans linking against an unstable version of Roslyn, which we cannot do for a stable package version

**Comment by ReubenBond** [comment]
Fixed in v3.5.1 https://github.com/dotnet/orleans/releases/tag/v3.5.1
